interfaces/builtin/connectivity_observe_test.go

```diff
@@ -0,0 +1,83 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/snap"
+)
+
+type ConnectivityObserveSuite struct {
+	iface interfaces.Interface
+	slot  *interfaces.Slot
+	plug  *interfaces.Plug
+}
+
+var _ = Suite(&ConnectivityObserveSuite{
+	iface: &builtin.ConnectivityObserveInterface{},
+	slot: &interfaces.Slot{
+		SlotInfo: &snap.SlotInfo{
+			Snap:      &snap.Info{SuggestedName: "server", Type: snap.TypeOS},
+			Name:      "connectivity-observe",
+			Interface: "connectivity-observe",
+		},
+	},
+	plug: &interfaces.Plug{
+		PlugInfo: &snap.PlugInfo{
+			Snap:      &snap.Info{SuggestedName: "client"},
+			Name:      "connectivity-observe",
+			Interface: "connectivity-observe",
+		},
+	},
+})
+
+func (s *ConnectivityObserveSuite) TestName(c *C) {
+	c.Check(s.iface.Name(), Equals, "connectivity-observe")
+}
+
+func (s *ConnectivityObserveSuite) TestSanitizeIncorrectInterface(c *C) {
+	c.Check(func() { s.iface.SanitizeSlot(&interfaces.Slot{SlotInfo: &snap.SlotInfo{Interface: "other"}}) },
+		PanicMatches, `slot is not of interface "connectivity-observe"`)
+	c.Check(func() { s.iface.SanitizePlug(&interfaces.Plug{PlugInfo: &snap.PlugInfo{Interface: "other"}}) },
+		PanicMatches, `plug is not of interface "connectivity-observe"`)
+}
+
+func (s *ConnectivityObserveSuite) TestUsedSecuritySystems(c *C) {
+	// connected slots have a non-nil security snippet for apparmor
+	snippet, err := s.iface.ConnectedSlotSnippet(s.plug, s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, Not(IsNil))
```

Comment:
I've added the following:
	c.Check(string(snippet), testutil.Contains, "interface=org.freedesktop.DBus.*")
	c.Check(string(snippet), testutil.Contains, "peer=(label=\"snap.client.*\"")

I'm not sure about this as jdstrand was suggesting: label=\"snap.connectivity-observe.*, but then running the tests I saw the label was different. 
Could this be an error from my side?